ambuda/models/auth.py

```diff
@@ -28,13 +28,34 @@ class User(AmbudaUserMixin, Base):
     #: The user's self-description.
     description = Column(Text_, nullable=False, default="")
 
+    #: If the user deleted their account.
+    is_deleted = Column(Boolean, nullable=False, default=False)
+
+    #: If the user was banned..
+    is_banned = Column(Boolean, nullable=False, default=False)
+
+    #: If the user has verified their email.
+    is_verified = Column(Boolean, nullable=False, default=False)
+
     #: All roles available for this user.
     roles = relationship("Role", secondary="user_roles")
 
     def set_password(self, raw_password: str):
         """Hash and save the given password."""
         self.password_hash = generate_password_hash(raw_password)
 
+    def set_is_deleted(self, is_deleted: bool):
```

Comment:
Why use setters instead of `user.is_deleted = True`?

Reply:
To be consistent with `set_password` as a property that is modified after-the-fact (and I'm biased with my previous Java experience haha). I can remove it if it's too verbose.

Reply:
`set_password` is kind of a special case to abstract away the password hashing logic. Otherwise I think plain attributes are more manageable long-term